### PR TITLE
fix: use "T" separator in time format placeholder to accomodate Safari

### LINKED ISSUE #3620: Time Ticks: Safari does not allow full RFC3339
Original issue was closed but a comment has brought to light an issue. See https://github.com/influxdata/ui/issues/3396#issuecomment-1015949880  

In Safari, using the suggested placeholder `YYYY-MM-DD HH:MM:SSZ` results in an invalid time, even though this format follows the suggested placeholder. Specfically a space is used instead of the `T` and Safari rejects the space as a valid separator.

Screen Shot 2022-01-18 at 5 08 54 PM

### PR DESCRIPTION
**Pull request by TCL735**
Closes #3620 

- uses `T` as the separator for the placeholder, reducing confusion for Safari users
- still allows space as the separator for other browsers that are more flexible such as Chrome and Firefox
- minor change to the case for the time portion of the format to differentiate `M` (month) and `m` (minutes)

This works in Safari:

https://user-images.githubusercontent.com/10736577/150049339-7923fd6b-6ead-487a-8c91-2ffa41db2d80.mp4

And it still works in Chrome using the space separator:

https://user-images.githubusercontent.com/10736577/150049484-263575dc-6ad3-4a6b-aae0-c2c982c3b837.mp4

### CODE REVIEW

File: src/utils/datetime/formatters.ts [hunk 1]
```diff
@@ -804,7 +804,7 @@ export const convertDateToRFC3339 = (date: Date, timeZone: string): string => {
     const localTime = timeStringParsed[0]
     const utcOffset = timeStringParsed[1].replace('GMT', '')
 
-    return `${year}-${month}-${dayOfMonth} ${localTime}${utcOffset}`
+    return `${year}-${month}-${dayOfMonth}T${localTime}${utcOffset}`
```
Comment (hoorayimhelping) [inline src/utils/datetime/formatters.ts:807]: is there a test case we can add to this?

**Review by hoorayimhelping — APPROVED** [review]
Looks good, question about needing a test case for this

File: src/utils/datetime/formatters.ts (hunk 1 above)
Comment (TCL735) [reply src/utils/datetime/formatters.ts:807]: Yes, just pushed it up as additional change.

**Review by hoorayimhelping — APPROVED** [review]
Nice tests!

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 51 | :red_circle:
accessibility | 97 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=3622)

**Pull request merged**